cms/djangoapps/contentstore/views/transcripts_ajax.py

```diff
@@ -319,68 +342,83 @@ def check_transcripts(request):  # lint-amnesty, pylint: disable=too-many-statem
+                    html5_subs.append(contentstore().find(content_location).data)
+                    transcripts_presence['html5_local'].append(html5_id)
+                except NotFoundError:
+                    log.debug("Can't find transcripts in storage for non-youtube video_id: %s", html5_id)
+                if len(html5_subs) == 2:  # check html5 transcripts for equality
+                    transcripts_presence['html5_equal'] = (
+                        json.loads(html5_subs[0].decode('utf-8')) == json.loads(html5_subs[1].decode('utf-8'))
                     )
-                    for (_, link) in transcript_links.items():
-                        youtube_server_subs = get_transcript_from_youtube(
-                            link, youtube_id, item.runtime.service(item, "i18n")
-                        )
-                        if json.loads(local_transcripts) == youtube_server_subs:  # check transcripts for equality
-                            transcripts_presence['youtube_diff'] = False
-                except GetTranscriptsFromYouTubeException:
-                    pass
-
-        # Check for html5 local transcripts presence
-        html5_subs = []
-        for html5_id in videos['html5']:
-            filename = f'subs_{html5_id}.srt.sjson'
-            content_location = StaticContent.compute_location(item.location.course_key, filename)
-            try:
-                html5_subs.append(contentstore().find(content_location).data)
-                transcripts_presence['html5_local'].append(html5_id)
-            except NotFoundError:
-                log.debug("Can't find transcripts in storage for non-youtube video_id: %s", html5_id)
-            if len(html5_subs) == 2:  # check html5 transcripts for equality
-                transcripts_presence['html5_equal'] = (
-                    json.loads(html5_subs[0].decode('utf-8')) == json.loads(html5_subs[1].decode('utf-8'))
-                )
 
         command, __ = _transcripts_logic(transcripts_presence, videos)
 
     transcripts_presence.update({'command': command})
     return JsonResponse(transcripts_presence)
 
 
+def _check_youtube_transcripts(transcripts_presence, youtube_id, item, isLibraryContent):
+    """
+    Check for youtube transcripts presence
+    """
+    transcripts_presence['is_youtube_mode'] = True
+
+    if get_transcript_link_from_youtube(youtube_id):
+        transcripts_presence['youtube_server'] = True
+
+    if not isLibraryContent:
+        # youtube local
+        filename = f'subs_{youtube_id}.srt.sjson'
+        content_location = StaticContent.compute_location(item.location.course_key, filename)
+        try:
+            local_transcripts = contentstore().find(content_location).data.decode('utf-8')
+            transcripts_presence['youtube_local'] = True
+        except NotFoundError:
+            log.debug("Can't find transcripts in storage for youtube id: %s", youtube_id)
+
+    #check youtube local and server transcripts for equality
```

Comment:
nit: pylint should have caught this?
```suggestion
    # check youtube local and server transcripts for equality
```

Reply:
Updated https://github.com/openedx/edx-platform/pull/36058/commits/590a5cda91f566801234e9f7f96b5f00dc553633